bpf/bpf_lxc.c

```diff
@@ -895,7 +895,11 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 skip_egress_gateway:
 #endif
 
-#ifdef ENABLE_VTEP
+	/* L7 proxy result in VTEP redirection in bpf_host, but when L7 proxy disabled
+	 * We want VTEP redirection handled earlier here to avoid packets passing to
+	 * stack to bpf_host for VTEP redirection.
+	 */
+#if defined(ENABLE_VTEP) && !defined(ENABLE_L7_PROXY)
```

Comment:
I'm a bit confused, isn't this the one path that guarantees that traffic from a pod will get routed to the VTEP correctly? Doesn't this change break VTEP support if you have enabled L7 proxy?

Reply:
it does not break VTEP support, `!defined(ENABLE_L7_PROXY)` was added when I was testing the VTEP redirection in bpf_host to make sure VTEP redirection works in bpf_host, Paul and I also noticed that when L7 proxy enabled, but no L7 policy applied to pod, the VTEP redirection actually happens in bpf_lxc. I do prefer your choice though to remove `!defined(ENABLE_L7_PROXY)` to guarantees the traffic from pod will get routed to VTEP when L7 proxy disabled, or when L7 proxy enabled but no L7 policy applied to pod.